windows: util.WriteTerminateMessage() panics always

## Checklist

* [x] Double-checked my configuration.
* [x] Tested using the latest version.
* [x] Used the Emissary executor.

## Summary

**What happened/what you expected to happen?**
When running failing workflows on windows workers, writing the termination log always fails because the directory `C:\dev` does not exist. `/dev/termination-log` gets mapped to `C:\dev\termination-log` in windows. 

https://github.com/argoproj/argo-workflows/blob/master/util/util.go#L53..L58

Because that function panics when writing is not possible, it panics always in windows. 

I have two possible fixes for this:
https://github.com/helio/argo-workflows/commit/fb304989479f046400fac37d5056027bfac09edd (different behaviour in windows vs unix - could also be done via a runtime.GOOS check)
https://github.com/helio/argo-workflows/commit/41b25e8ac0ab96a0ee3a844c9f1a2b6b7144807a (do not panic in any case)

I'm not sure why there is a panic when writing to that file does not work. 

Calling sites (x marks if that should work with no panic):
- [x] https://github.com/argoproj/argo-workflows/blob/master/cmd/argoexec/main.go#L24..L26
- [x] https://github.com/argoproj/argo-workflows/blob/master/cmd/argoexec/commands/root.go#L158..L159
- [x] https://github.com/argoproj/argo-workflows/blob/master/workflow/executor/executor.go#L150..L157 (at least in the first case, but calls to that function should be fine in any case)
- [x] https://github.com/argoproj/argo-workflows/blob/master/workflow/executor/executor.go#L1060..L1061 (I believe a panic is even harmful in this case)

**What version are you running?**
v3.3.1

## Diagnostics

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: hello-world-
  namespace: default
  labels:
    workflows.argoproj.io/archive-strategy: "false"
spec:
  entrypoint: whalesay
  templates:
    - name: whalesay
      container:
        image: mcr.microsoft.com/windows/nanoserver:1809
        command: ["powershell", "-Command"] # powershell is not in nanoserver, making this fail immediately
        args: ["echo hello world"]
      nodeSelector:
        kubernetes.io/os: windows
```

```bash
# Logs from the workflow controller:
time="2022-03-23T11:50:42.269Z" level=info msg="Processing workflow" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:42.274Z" level=info msg="Updated phase  -> Running" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:42.274Z" level=info msg="Pod node hello-world-5z6hr initialized Pending" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:42.294Z" level=info msg="Created pod: hello-world-5z6hr (hello-world-5z6hr)" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:42.294Z" level=info msg="TaskSet Reconciliation" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:42.294Z" level=info msg=reconcileAgentPod namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:42.333Z" level=info msg="Workflow update successful" namespace=default phase=Running resourceVersion=8402096 workflow=hello-world-5z6hr
time="2022-03-23T11:50:52.295Z" level=info msg="Processing workflow" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:52.296Z" level=info msg="node changed" new.message=PodInitializing new.phase=Pending new.progress=0/1 nodeID=hello-world-5z6hr old.message= old.phase=Pending old.progress=0/1
time="2022-03-23T11:50:52.296Z" level=info msg="TaskSet Reconciliation" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:52.296Z" level=info msg=reconcileAgentPod namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:50:52.308Z" level=info msg="Workflow update successful" namespace=default phase=Running resourceVersion=8402177 workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.306Z" level=info msg="Processing workflow" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.306Z" level=info msg="node changed" new.message="Error (exit code 2)" new.phase=Failed new.progress=0/1 nodeID=hello-world-5z6hr old.message=PodInitializing old.phase=Pending old.progress=0/1
time="2022-03-23T11:51:02.306Z" level=info msg="TaskSet Reconciliation" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.306Z" level=info msg=reconcileAgentPod namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.306Z" level=info msg="Updated phase Running -> Failed" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.306Z" level=info msg="Updated message  -> Error (exit code 2)" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.306Z" level=info msg="Marking workflow completed" namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.306Z" level=info msg="Checking daemoned children of " namespace=default workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.312Z" level=info msg="cleaning up pod" action=deletePod key=default/hello-world-5z6hr-1340600742-agent/deletePod
time="2022-03-23T11:51:02.324Z" level=info msg="Workflow update successful" namespace=default phase=Failed resourceVersion=8402263 workflow=hello-world-5z6hr
time="2022-03-23T11:51:02.340Z" level=info msg="cleaning up pod" action=labelPodCompleted key=default/hello-world-5z6hr/labelPodCompleted


# If the workflow's pods have not been created, you can skip the rest of the diagnostics.

# The workflow's pods that are problematic:
apiVersion: v1
items:
- apiVersion: v1
  kind: Pod
  metadata:
    annotations:
      kubectl.kubernetes.io/default-container: main
      workflows.argoproj.io/node-id: hello-world-5z6hr
      workflows.argoproj.io/node-name: hello-world-5z6hr
    creationTimestamp: "2022-03-23T11:50:42Z"
    labels:
      workflows.argoproj.io/completed: "true"
      workflows.argoproj.io/workflow: hello-world-5z6hr
    name: hello-world-5z6hr
    namespace: default
    ownerReferences:
    - apiVersion: argoproj.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: Workflow
      name: hello-world-5z6hr
      uid: 84333e4d-eefd-426a-a75b-968d025a32c1
    resourceVersion: "8402266"
    uid: 1ee87c80-ca7a-4392-b84a-7d0a68268953
  spec:
    containers:
    - command:
      - argoexec
      - wait
      - --loglevel
      - info
      env:
      - name: ARGO_POD_NAME
        valueFrom:
          fieldRef:
            apiVersion: v1
            fieldPath: metadata.name
      - name: ARGO_CONTAINER_RUNTIME_EXECUTOR
        value: emissary
      - name: GODEBUG
        value: x509ignoreCN=0
      - name: ARGO_WORKFLOW_NAME
        value: hello-world-5z6hr
      - name: ARGO_WORKFLOW_UID
        value: 84333e4d-eefd-426a-a75b-968d025a32c1
      - name: ARGO_CONTAINER_NAME
        value: wait
      - name: ARGO_TEMPLATE
        value: '{"name":"whalesay","inputs":{},"outputs":{},"nodeSelector":{"kubernetes.io/os":"windows"},"metadata":{},"container":{"name":"","image":"mcr.microsoft.com/windows/nanoserver:1809","command":["powershell","-Command"],"args":["echo
          hello world"],"resources":{}},"tolerations":[{"key":"node.helio.exchange/role","value":"render","effect":"NoSchedule"}]}'
      - name: ARGO_NODE_ID
        value: hello-world-5z6hr
      - name: ARGO_INCLUDE_SCRIPT_OUTPUT
        value: "false"
      - name: ARGO_DEADLINE
        value: "0001-01-01T00:00:00Z"
      - name: ARGO_PROGRESS_FILE
        value: /var/run/argo/progress
      - name: ARGO_PROGRESS_PATCH_TICK_DURATION
        value: 1m0s
      - name: ARGO_PROGRESS_FILE_TICK_DURATION
        value: 3s
      image: quay.io/argoproj/argoexec:v3.3.1
      imagePullPolicy: IfNotPresent
      name: wait
      resources: {}
      terminationMessagePath: /dev/termination-log
      terminationMessagePolicy: File
      volumeMounts:
      - mountPath: /var/run/argo
        name: var-run-argo
      - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
        name: kube-api-access-w8vbc
        readOnly: true
    - args:
      - echo hello world
      command:
      - /var/run/argo/argoexec
      - emissary
      - --
      - powershell
      - -Command
      env:
      - name: ARGO_CONTAINER_NAME
        value: main
      - name: ARGO_TEMPLATE
        value: '{"name":"whalesay","inputs":{},"outputs":{},"nodeSelector":{"kubernetes.io/os":"windows"},"metadata":{},"container":{"name":"","image":"mcr.microsoft.com/windows/nanoserver:1809","command":["powershell","-Command"],"args":["echo
          hello world"],"resources":{}},"tolerations":[{"key":"node.helio.exchange/role","value":"render","effect":"NoSchedule"}]}'
      - name: ARGO_NODE_ID
        value: hello-world-5z6hr
      - name: ARGO_INCLUDE_SCRIPT_OUTPUT
        value: "false"
      - name: ARGO_DEADLINE
        value: "0001-01-01T00:00:00Z"
      - name: ARGO_PROGRESS_FILE
        value: /var/run/argo/progress
      - name: ARGO_PROGRESS_PATCH_TICK_DURATION
        value: 1m0s
      - name: ARGO_PROGRESS_FILE_TICK_DURATION
        value: 3s
      image: mcr.microsoft.com/windows/nanoserver:1809
      imagePullPolicy: IfNotPresent
      name: main
      resources: {}
      terminationMessagePath: /dev/termination-log
      terminationMessagePolicy: File
      volumeMounts:
      - mountPath: /var/run/argo
        name: var-run-argo
      - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
        name: kube-api-access-w8vbc
        readOnly: true
    dnsPolicy: ClusterFirst
    enableServiceLinks: true
    initContainers:
    - command:
      - argoexec
      - init
      - --loglevel
      - info
      env:
      - name: ARGO_POD_NAME
        valueFrom:
          fieldRef:
            apiVersion: v1
            fieldPath: metadata.name
      - name: ARGO_CONTAINER_RUNTIME_EXECUTOR
        value: emissary
      - name: GODEBUG
        value: x509ignoreCN=0
      - name: ARGO_WORKFLOW_NAME
        value: hello-world-5z6hr
      - name: ARGO_WORKFLOW_UID
        value: 84333e4d-eefd-426a-a75b-968d025a32c1
      - name: ARGO_CONTAINER_NAME
        value: init
      - name: ARGO_TEMPLATE
        value: '{"name":"whalesay","inputs":{},"outputs":{},"nodeSelector":{"kubernetes.io/os":"windows"},"metadata":{},"container":{"name":"","image":"mcr.microsoft.com/windows/nanoserver:1809","command":["powershell","-Command"],"args":["echo
          hello world"],"resources":{}},"tolerations":[{"key":"node.helio.exchange/role","value":"render","effect":"NoSchedule"}]}'
      - name: ARGO_NODE_ID
        value: hello-world-5z6hr
      - name: ARGO_INCLUDE_SCRIPT_OUTPUT
        value: "false"
      - name: ARGO_DEADLINE
        value: "0001-01-01T00:00:00Z"
      - name: ARGO_PROGRESS_FILE
        value: /var/run/argo/progress
      - name: ARGO_PROGRESS_PATCH_TICK_DURATION
        value: 1m0s
      - name: ARGO_PROGRESS_FILE_TICK_DURATION
        value: 3s
      image: quay.io/argoproj/argoexec:v3.3.1
      imagePullPolicy: IfNotPresent
      name: init
      resources: {}
      terminationMessagePath: /dev/termination-log
      terminationMessagePolicy: File
      volumeMounts:
      - mountPath: /var/run/argo
        name: var-run-argo
      - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
        name: kube-api-access-w8vbc
        readOnly: true
    nodeName: REDACTED
    nodeSelector:
      kubernetes.io/os: windows
    preemptionPolicy: PreemptLowerPriority
    priority: 0
    restartPolicy: Never
    schedulerName: default-scheduler
    securityContext: {}
    serviceAccount: default
    serviceAccountName: default
    terminationGracePeriodSeconds: 30
    tolerations:
    - effect: NoExecute
      key: node.kubernetes.io/not-ready
      operator: Exists
      tolerationSeconds: 300
    - effect: NoExecute
      key: node.kubernetes.io/unreachable
      operator: Exists
      tolerationSeconds: 300
    volumes:
    - emptyDir: {}
      name: var-run-argo
    - name: kube-api-access-w8vbc
      projected:
        defaultMode: 420
        sources:
        - serviceAccountToken:
            expirationSeconds: 3607
            path: token
        - configMap:
            items:
            - key: ca.crt
              path: ca.crt
            name: kube-root-ca.crt
        - downwardAPI:
            items:
            - fieldRef:
                apiVersion: v1
                fieldPath: metadata.namespace
              path: namespace
  status:
    conditions:
    - lastProbeTime: null
      lastTransitionTime: "2022-03-23T11:50:53Z"
      status: "True"
      type: Initialized
    - lastProbeTime: null
      lastTransitionTime: "2022-03-23T11:50:41Z"
      message: 'containers with unready status: [wait main]'
      reason: ContainersNotReady
      status: "False"
      type: Ready
    - lastProbeTime: null
      lastTransitionTime: "2022-03-23T11:50:41Z"
      message: 'containers with unready status: [wait main]'
      reason: ContainersNotReady
      status: "False"
      type: ContainersReady
    - lastProbeTime: null
      lastTransitionTime: "2022-03-23T11:50:42Z"
      status: "True"
      type: PodScheduled
    containerStatuses:
    - containerID: docker://825853b984dd017cde1d958d21044a2d9df58a3c17b5eee7a1c2cd185a816ab8
      image: mcr.microsoft.com/windows/nanoserver:1809
      imageID: docker-pullable://mcr.microsoft.com/windows/nanoserver@sha256:e508123972ca40c7662658ee1893a8d20f44a9dcd49be3ffa9dec559e525b71d
      lastState: {}
      name: main
      ready: false
      restartCount: 0
      started: false
      state:
        terminated:
          containerID: docker://825853b984dd017cde1d958d21044a2d9df58a3c17b5eee7a1c2cd185a816ab8
          exitCode: 2
          finishedAt: "2022-03-23T11:50:55Z"
          reason: Error
          startedAt: "2022-03-23T11:50:55Z"
    - containerID: docker://d45c2fedb39bc742a1470e2760ce10b17b0ecb392868eb225d4480997ae94106
      image: quay.io/argoproj/argoexec:v3.3.1
      imageID: docker-pullable://quay.io/argoproj/argoexec@sha256:a7ced0229f6285611091ba62633dc45661f47ebcc692e67af6234fd2c2f55cb3
      lastState: {}
      name: wait
      ready: false
      restartCount: 0
      started: false
      state:
        terminated:
          containerID: docker://d45c2fedb39bc742a1470e2760ce10b17b0ecb392868eb225d4480997ae94106
          exitCode: 0
          finishedAt: "2022-03-23T11:50:55Z"
          reason: Completed
          startedAt: "2022-03-23T11:50:54Z"
    hostIP: 10.0.31.184
    initContainerStatuses:
    - containerID: docker://ea011ce476ab93d22f98e484a0ff30c25f34e4aa7dff0353ce54c8067d9e8d37
      image: quay.io/argoproj/argoexec:v3.3.1
      imageID: docker-pullable://quay.io/argoproj/argoexec@sha256:a7ced0229f6285611091ba62633dc45661f47ebcc692e67af6234fd2c2f55cb3
      lastState: {}
      name: init
      ready: true
      restartCount: 0
      state:
        terminated:
          containerID: docker://ea011ce476ab93d22f98e484a0ff30c25f34e4aa7dff0353ce54c8067d9e8d37
          exitCode: 0
          finishedAt: "2022-03-23T11:50:52Z"
          reason: Completed
          startedAt: "2022-03-23T11:50:47Z"
    phase: Failed
    podIP: 192.168.52.6
    podIPs:
    - ip: 192.168.52.6
    qosClass: BestEffort
    startTime: "2022-03-23T11:50:41Z"
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

# Logs from in your workflow's wait container, something like:
time="2022-03-23T11:50:54.516Z" level=info msg="Creating a emissary executor"
time="2022-03-23T11:50:54.519Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2022-03-23T11:50:54.519Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=default podName=hello-world-5z6hr template="{\"name\":\"whalesay\",\"inputs\":{},\"outputs\":{},\"nodeSelector\":{\"kubernetes.io/os\":\"windows\"},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"mcr.microsoft.com/windows/nanoserver:1809\",\"command\":[\"powershell\",\"-Command\"],\"args\":[\"echo hello world\"],\"resources\":{}},\"tolerations\":[{\"key\":\"node.helio.exchange/role\",\"value\":\"render\",\"effect\":\"NoSchedule\"}]}" version="&Version{Version:v3.3.1+3d60693.dirty,BuildDate:2022-03-18T18:31:02Z,GitCommit:3d606939dcc78d22ff5fa1401703c619639d019f,GitTag:v3.3.1,GitTreeState:dirty,GoVersion:go1.17,Compiler:gc,Platform:windows/amd64,}"
time="2022-03-23T11:50:54.526Z" level=info msg="Starting deadline monitor"
time="2022-03-23T11:50:55.526Z" level=info msg="Main container completed"
time="2022-03-23T11:50:55.526Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2022-03-23T11:50:55.527Z" level=info msg="No output parameters"
time="2022-03-23T11:50:55.527Z" level=info msg="No output artifacts"
time="2022-03-23T11:50:55.528Z" level=info msg="Killing sidecars []"
time="2022-03-23T11:50:55.528Z" level=info msg="Alloc=9085 TotalAlloc=10609 Sys=20576 NumGC=1 Goroutines=7"

# Logs from pod
Error: failed to find name in PATH: exec: "powershell": executable file not found in %PATH%
panic: open /dev/termination-log: The system cannot find the path specified.

goroutine 1 [running]:
github.com/argoproj/argo-workflows/v3/util.WriteTerminateMessage(...)
        C:/Users/ContainerAdministrator/go/src/github.com/argoproj/argo-workflows/util/util.go:56
main.main()
        C:/Users/ContainerAdministrator/go/src/github.com/argoproj/argo-workflows/cmd/argoexec/main.go:24 +0x12e
```

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

--- a/util/util.go
+++ b/util/util.go
@@ -53,7 +53,7 @@ func GetSecrets(ctx context.Context, clientSet kubernetes.Interface, namespace, 
 func WriteTerminateMessage(message string) {
 	err := ioutil.WriteFile("/dev/termination-log", []byte(message), 0o600)
 	if err != nil {
-		panic(err)
+		println("unable to write termination log: " + err.Error())
 	}
 }
 

